File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02958/p02958_num6_parsed.py
# Given input "N" and sequence "p"
N = int(input())
p = list(map(int, input().split()))

# Check if the sequence is already sorted in ascending order
if p == sorted(p):
    print("YES")
    exit()

# Iterate through each index of the sequence
for i in range(N):
    # For each index, iterate through all indices after it
    for j in range(i+1, N):
        # Swap p[i] and p[j]
        p[i], p[j] = p[j], p[i]
        
        # Check if the sequence is now sorted in ascending order
        if p == sorted(p):
            print("YES")
            exit()
        
        # Swap back the elements
        p[i], p[j] = p[j], p[i]

# If the sequence cannot be sorted by performing the operation once, print "NO"
print("NO")

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02958/p02958_num6_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> p(N);
    for (int i = 0; i < N; i++) {
        cin >> p[i];
    }

    // Check if the given permutation is already sorted
    bool alreadySorted = true;
    for (int i = 0; i < N; i++) {
        if (p[i] != i + 1) {
            alreadySorted = false;
            break;
        }
    }

    // If already sorted, print YES
    if (alreadySorted) {
        cout << "YES" << endl;
        return 0;
    }

    // Find the two positions i and j where p[i] > p[j]
    int i = -1, j = -1;
    for (int k = 0; k < N; k++) {
        if (p[k] > p[k + 1]) {
            i = k;
            break;
        }
    }
    for (int k = N - 1; k > 0; k--) {
        if (p[k] < p[k - 1]) {
            j = k;
            break;
        }
    }

    // Swap p[i] and p[j] to check if it becomes sorted
    swap(p[i], p[j]);

    // Check if the permutation is sorted now
    bool sorted = true;
    for (int k = 0; k < N; k++) {
        if (p[k] != k + 1) {
            sorted = false;
            break;
        }
    }

    // If sorted, print YES, otherwise print NO
    if (sorted) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}